### PR TITLE
ov 2.0: python image_inpainting_demo

### PR DESCRIPTION
**Pull request by anzhella-pankratova**
Migrate python image_inpainting_demo to OV2.0 API

### CODE REVIEW

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Comment by vladimir-dudnik** [comment]
@anzhella-pankratova 01:19:48  RuntimeError: Attempt to get a name for a Tensor without names
either we did something wrong in new API or there are issues with API itself

cc @AlexeyLebedev1 @AnastasiaKazantaeva

**Comment by anzhella-pankratova** [comment]
> @anzhella-pankratova 01:19:48 RuntimeError: Attempt to get a name for a Tensor without names either we did something wrong in new API or there are issues with API itself
> 
> cc @AlexeyLebedev1 @AnastasiaKazantaeva

I successfully have launched the demo locally with IR version 10, however with IR version 11 this error is raised. So the problem is not caused by new API, the Model Optimizer converts the model with tensors without names

**Comment by vladimir-dudnik** [comment]
@andrei-kochin is it known issue?

**Comment by andrei-kochin** [comment]
> @andrei-kochin is it known issue?

Yes it is known issue with missing tensor names

File: demos/image_inpainting_demo/python/inpainting.py [hunk 1]
```diff
@@ -15,32 +15,36 @@
...
-        assert channels == 3, "Expected 3-channel input"
+        _, channels, input_height, input_width = model.input(self.image_input_layer).shape
+        if channels != 3:
+            raise RuntimeError("The model expects 3 channels for {} input layer".format(self.image_input_layer))
 
-        _, channels, mask_height, mask_width = model.input_info[self._input_layer_names[1]].input_data.shape
-        assert channels == 1, "Expected 1-channel input"
+        _, channels, mask_height, mask_width = model.input(self.mask_input_layer).shape
+        if channels != 1:
+            raise RuntimeError("The model expects 3 channels for {} input layer".format(self.mask_input_layer))
```
Comment (eaidova) [inline demos/image_inpainting_demo/python/inpainting.py:37]: mask is single channel input
```suggestion
            raise RuntimeError("The model expects 1 channels for {} input layer".format(self.mask_input_layer))
```

Comment (anzhella-pankratova) [reply demos/image_inpainting_demo/python/inpainting.py:37]: Thanks, it was a typo

**Comment by eaidova** [comment]
> > @andrei-kochin is it known issue?
> 
> Yes it is known issue with missing tensor names

**Comment by eaidova** [comment]
@

> > > @andrei-kochin is it known issue?
> > 
> > 
> > Yes it is known issue with missing tensor names

@andrei-kochin  @vladimir-dudnik  should we wait for fix or we can workaround the issue (using indexes instead of tensor names for inference)?

**Comment by anzhella-pankratova** [comment]
Jenkins please retry a build

**Comment by anzhella-pankratova** [comment]
Jenkins please retry a build

**Comment by eaidova** [comment]
`RuntimeError: Can't SetBlob with name: Placeholder, because model input and blob are incompatible`

@anzhella-pankratova
looks like something wrong with data shape (old api automatically fit data to input shape, if it has the same number of elements, but different shape representation (e.g. input (1, 1, 10, 10) and data (10, 10) - ok), new api expects that you provide data with the same shape

**Comment by anzhella-pankratova** [comment]
> `RuntimeError: Can't SetBlob with name: Placeholder, because model input and blob are incompatible`
> 
> @anzhella-pankratova looks like something wrong with data shape (old api automatically fit data to input shape, if it has the same number of elements, but different shape representation (e.g. input (1, 1, 10, 10) and data (10, 10) - ok), new api expects that you provide data with the same shape

I've updated the image preprocessing step with NHWC support, now it's fine.

File: demos/image_inpainting_demo/python/inpainting.py [hunk 2]
```diff
@@ -15,40 +15,49 @@
...
+            raise RuntimeError("Mask size is expected to be equal to image size")
         self.input_height = input_height
         self.input_width = input_width
 
 
     def infer(self, image, mask):
-        output = self._exec_model.infer(inputs={self._input_layer_names[0]: image, self._input_layer_names[1]: mask})
-        return output[self._output_layer_name]
+        output = self.infer_request.infer(inputs={self.image_input_layer: image, self.mask_input_layer: mask})
+        return next(iter(output.values()))
```
Comment (eaidova) [inline demos/image_inpainting_demo/python/inpainting.py:50]: do you plan unify logic with output[self.output_tensor] later for all demos for models with single output?

Comment (anzhella-pankratova) [reply demos/image_inpainting_demo/python/inpainting.py:50]: Yeah, I will change it in all demos in a separate PR

**Review by eaidova — APPROVED** [review]

**Pull request merged**